js/genform.js

```diff
@@ -180,3 +181,16 @@ function checkContactValidity(el) {
         el.classList.add('is-danger');
     }
 }
+
+function autoprefixEmail(el) {
+    // Add mailto: prefix, e.g. contact@example.com -> mailto:contact@example.com
+
+    // NOTE: If this regular expression is changed in future, it should always fail if
+    // the user enters some other schema (e.g. https://) so they always have the option of overriding it.
+    // Currently this is achieved by not allowing a colon to appear in the input.
+    if (/^[a-z0-9_+.-]+@[a-z0-9_+.-]+$/i.test(el.value)) {
```

Comment:
> and contain only alphanumeric characters

Note that emails should usually never be validated like this. Also all RegEx'es to validate will eventually fail, and if you use emojis in your domain name or so… It is [thus usually hard or even discouraged to use RegEx for that](https://stackoverflow.com/q/46155/5008962)…

See e.g. https://www.emailregex.com/

Reply:
You're right, and the RegEx should potentially be improved over time - but this is a convenience feature and if the RegEx fails to match then the user can insert the `mailto:` themselves. What would be much worse is if `mailto:` is prepended to non-emails I think